src/clientsettingsdlg.cpp

```diff
@@ -1063,25 +1063,31 @@ void CClientSettingsDlg::OnSndCrdBufferDelayButtonGroupClicked ( QAbstractButton
     UpdateDisplay();
 }
 
+/// @method
+/// @brief Sets upstream rate label to current upload rate if the client is connected, else resets label
 void CClientSettingsDlg::UpdateUploadRate()
 {
-    // update upstream rate information label
-    lblUpstreamValue->setText ( QString().setNum ( pClient->GetUploadRateKbps() ) );
-    lblUpstreamUnit->setText ( "kbps" );
+    // update upstream rate information label if needed
+    if ( pClient->IsConnected() )
+    {
+        lblUpstreamValue->setText ( QString().setNum ( pClient->GetUploadRateKbps() ) );
```

Comment:
This is
```
int GetUploadRateKbps() { return Channel.GetUploadRateKbps(); }
```
So it makes far more sense to check the `Channel.IsConnected()`.

But really, it's down to the audio settings, nothing to do with whether a sound card is connected or a server is connected:
![image](https://github.com/user-attachments/assets/d9a71ea7-eefb-4c71-972b-1e6d96e155be)
It can be computed from those values, I think.  (But it's probably the Channel that gets passed the values to do the computation - it gets passed uncompressed audio and does the number crunching.)